core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -138,13 +152,57 @@ public void onDestroyableBreak(DestroyableHealthChangeEvent event) {
   }
 
   @EventHandler(priority = EventPriority.MONITOR)
-  public void onFlagCapture(FlagCaptureEvent event) {
-    getPlayerStat(event.getCarrier()).onFlagCapture();
+  public void onMonumentDestroy(DestroyableDestroyedEvent event) {
+    event
+        .getDestroyable()
+        .getContributions()
+        .forEach(
+            destroyer -> {
+              if (destroyer.getPlayerState() != null) {
+                getPlayerStat(destroyer.getPlayerState()).onMonumentDestroyed();
+              }
+            });
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void onCoreLeak(CoreLeakEvent event) {
+    event
+        .getCore()
+        .getContributions()
+        .forEach(
+            leaker -> {
+              if (leaker.getPlayerState() != null) {
+                // TODO: wrong method
+                getPlayerStat(leaker.getPlayerState()).onCoreLeak();
+              }
+            });
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR)
+  public void onGoalTouch(GoalTouchEvent event) {
+    if (event.getPlayer() == null || !event.getPlayer().getPlayer().isPresent()) return;
+
+    if (event.getGoal() instanceof MonumentWool) {
+      if (event.isFirstForPlayer()) {
+        getPlayerStat(event.getPlayer()).onWoolTouch();
+      }
+    }
+
+    if (event.getGoal() instanceof Flag) {
+      getPlayerStat(event.getPlayer()).onFlagPickup(event.isFirstForPlayer());
+    }
   }
 
   @EventHandler(priority = EventPriority.MONITOR)
-  public void onFlagHold(FlagPickupEvent event) {
-    getPlayerStat(event.getCarrier()).onFlagPickup();
+  public void onWoolCapture(PlayerWoolPlaceEvent event) {
+    if (event.getPlayer() != null && event.getPlayer().getPlayer().isPresent()) {
+      getPlayerStat(event.getPlayer()).onWoolCapture();
+    }
```

Comment:
is this about getting the player off of a player state? If so, that's not needed, keep it as:
```suggestion
    if (event.getPlayer() != null) {
      getPlayerStat(event.getPlayer()).onWoolCapture();
    }
```